src/Button.tsx

```diff
@@ -106,7 +106,7 @@ const Button: Button = React.forwardRef(
       className,
       prefix,
       active && 'active',
-      `${prefix}-${variant}`,
+      variant && `${prefix}-${variant}`,
```

Comment:
this is definitely a bug fix, i would split it into another PR with just this change so that it doesn't get locked up in the discussion around the other feature

Reply:
For sure, will do - thanks for the feedback.  Here we go: #5458.  Will remove this change from this PR shortly and force push.